spotify_to_ytmusic/ytmusic.py

```diff
@@ -33,17 +35,37 @@ def create_playlist(self, name, info, privacy="PRIVATE", tracks=None):
 
     def rate_song(self, id, rating):
         return self.api.rate_song(id, rating)
+    
+    def load_lookup_table(self):
+        try:
+            with open(path + "lookup.json") as f:
+                return json.load(f)
+        except FileNotFoundError:
+            return {}
+
+    def save_to_lookup_table(self, table):
+        with open(path + "lookup.json", 'w', encoding="utf-8") as f:
+            json.dump(table, f, ensure_ascii=False)
 
-    def search_songs(self, tracks):
+    def search_songs(self, tracks, use_cached: bool = False):
```

Comment:
Please add a test that checks whether the cache file has been created correctly when --use-cached is used.